### ISSUE TITLE
[BUG] TensorDict print output seems to be incorrect

### ISSUE THREAD

**Issue by ezhang7423** [facebookresearch/rl#367]
## Describe the bug

When creating a nested tensordict, indexing into the larger dict prints the wrong shape.
## To Reproduce

Steps to reproduce the behavior.

```
 from torchrl.data import TensorDict

z = TensorDict({str(k): TensorDict({k: torch.arange(300).reshape(150, 2) for k in string.ascii_lowercase}, batch_size=[150, 2]) for k in [0, 1, 2, 3, 4]}, batch_size=[150])`
print(z[0])
TensorDict(
    fields={
        0: TensorDict(
            fields={
                a: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                b: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                c: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                d: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                e: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                f: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                g: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                h: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                i: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                j: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                k: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                l: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                m: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                n: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                o: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                p: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                q: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                r: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                s: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                t: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                u: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                v: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                w: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                x: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                y: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                z: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64)},
            batch_size=torch.Size([150, 2]),
            device=cpu,
            is_shared=False),
        1: TensorDict(
            fields={
                a: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                b: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                c: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                d: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                e: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                f: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                g: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                h: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                i: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                j: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                k: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                l: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                m: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                n: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                o: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                p: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                q: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                r: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                s: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                t: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                u: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                v: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                w: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                x: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                y: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                z: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64)},
            batch_size=torch.Size([150, 2]),
            device=cpu,
            is_shared=False),
        2: TensorDict(
            fields={
                a: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                b: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                c: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                d: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                e: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                f: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                g: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                h: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                i: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                j: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                k: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                l: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                m: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                n: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                o: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                p: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                q: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                r: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                s: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                t: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                u: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                v: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                w: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                x: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                y: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                z: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64)},
            batch_size=torch.Size([150, 2]),
            device=cpu,
            is_shared=False),
        3: TensorDict(
            fields={
                a: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                b: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                c: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                d: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                e: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                f: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                g: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                h: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                i: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                j: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                k: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                l: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                m: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                n: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                o: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                p: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                q: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                r: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                s: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                t: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                u: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                v: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                w: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                x: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                y: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                z: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64)},
            batch_size=torch.Size([150, 2]),
            device=cpu,
            is_shared=False),
        4: TensorDict(
            fields={
                a: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                b: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                c: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                d: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                e: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                f: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                g: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                h: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                i: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                j: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                k: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                l: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                m: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                n: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                o: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                p: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                q: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                r: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                s: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                t: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                u: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                v: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                w: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                x: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                y: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64),
                z: Tensor(torch.Size([150, 2, 1]), dtype=torch.int64)},
            batch_size=torch.Size([150, 2]),
            device=cpu,
            is_shared=False)},
    batch_size=torch.Size([]),
    device=cpu,
    is_shared=False)
```

## Screenshots

I expect the batch_size for each key to be `[2]`.

## System info

Describe the characteristic of your environment:
 * Describe how the library was installed (pip, source, ...)
 * Python version
 * Versions of any other relevant libraries

```python

In [24]: import torchrl, numpy, sys
    ...: print(torchrl.__version__, numpy.__version__, sys.version, sys.platform)
0.0.1c0 1.23.1 3.8.5 (default, Sep  4 2020, 07:30:14)
[GCC 7.3.0] linux

```

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have read the [documentation](https://github.com/facebookresearch/rl/tree/main/docs/) (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)

**Comment by vmoens** [comment]
Thanks for reporting @ezhang7423 
This is due to the MetaTensor class passing the __repr__ result to the indexed MetaTensor when the MetaTensor points to a nested TensorDict ([here](https://github.com/facebookresearch/rl/blob/34fc9c65c317ec662a4975f323f67a3a0aa3425f/torchrl/data/tensordict/metatensor.py#L196)).
The easiest solution would probably be to pass a None for the `_repr_tensordict` arg and have the `TensorDict.__repr__` fall back on the real `TensorDict` for printing (rather than a cached value)

**Comment by vmoens** [comment]
Solved as of #370
